src/main/java/org/openrewrite/java/spring/security5/UseNewRequestMatchers.java

```diff
@@ -57,46 +60,28 @@ public String getDescription() {
-            }
-        });
-
-    }
-
-    private J.MethodInvocation maybeChangeMethodInvocation(J.MethodInvocation mi) {
-        return findRequestMatchersMethodWithMatchingParameterTypes(mi)
-                .map(requestMatchersMethod -> mi
-                        .withMethodType(requestMatchersMethod)
-                        .withName(mi.getName().withSimpleName("requestMatchers")))
-                .orElse(mi);
-    }
-
-    private Optional<JavaType.Method> findRequestMatchersMethodWithMatchingParameterTypes(J.MethodInvocation mi) {
-        JavaType.Method methodType = mi.getMethodType();
-        if (methodType == null) {
-            return Optional.empty();
-        }
-        List<JavaType> parameterTypes = methodType.getParameterTypes();
-        List<JavaType.Method> methods;
-        boolean isOverride = TypeUtils.isOverride(mi.getMethodType());
-        if (isOverride) {
-            methods = requireNonNull(methodType.getDeclaringType().getSupertype(), "superType").getMethods();
-        } else {
-            methods = methodType.getDeclaringType().getMethods();
-        }
-        return methods.stream()
-                .filter(m -> m.getName().equals("requestMatchers"))
-                .filter(m -> m.getParameterTypes().equals(parameterTypes))
-                .findFirst();
+                        new UsesMethod<>(ANT_MATCHERS),
+                        new UsesMethod<>(MVC_MATCHERS),
+                        new UsesMethod<>(REGEX_MATCHERS)),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                        J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                        boolean isCsrfMatcher = CSRF_MATCHERS.matches(mi);
+                        if ((ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi) || isCsrfMatcher)
+                                && mi.getSelect() != null) {
+                            String parametersTemplate = mi.getArguments().stream().map(arg -> "#{any()}").collect(joining(", "));
+                            List<Expression> parameters = new ArrayList<>();
+                            parameters.add(mi.getSelect());
+                            parameters.addAll(mi.getArguments());
+                            String replacementMethodName = isCsrfMatcher ? "ignoringRequestMatchers" : "requestMatchers";
+                            JavaTemplate template = JavaTemplate.builder(String.format("#{any()}\n." + replacementMethodName + "(%s)", parametersTemplate))
+                                    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "spring-security-config-5.8"))
+                                    .build();
+                            return template.apply(getCursor(), mi.getCoordinates().replace(), parameters.toArray());
```

Comment:
Briefly tried `mi.getCoordinates().replaceMethod()` here but that breaks the multiple replacements test case; not sure why. 🤔

Reply:
Possibly a templating bug

Reply:
Also [see thread above](https://github.com/openrewrite/rewrite-spring/pull/398#discussion_r1276855502).

Reply:
Now using `replaceMethod` which required the addition of `withSelect` and `withName`, but otherwise better now.